pkg/api/endpoints/util.go

```diff
@@ -34,42 +34,24 @@ import (
 // form for things like comparison.  The result is a newly allocated slice.
 func RepackSubsets(subsets []api.EndpointSubset) []api.EndpointSubset {
 	// First map each unique port definition to the sets of hosts that
-	// offer it.  The sets of hosts must be de-duped, using IP+UID as the key.
-	type addressKey struct {
-		ip  string
-		uid types.UID
-	}
+	// offer it.
 	allAddrs := map[addressKey]*api.EndpointAddress{}
 	portsToAddrs := map[api.EndpointPort]addressSet{}
 	for i := range subsets {
 		for j := range subsets[i].Ports {
-			epp := &subsets[i].Ports[j]
+			epp := subsets[i].Ports[j]
 			for k := range subsets[i].Addresses {
-				epa := &subsets[i].Addresses[k]
-				ak := addressKey{ip: epa.IP}
-				if epa.TargetRef != nil {
-					ak.uid = epa.TargetRef.UID
-				}
-				// Accumulate the address.
-				if allAddrs[ak] == nil {
-					// Make a copy so we don't write to the
-					// input args of this function.
-					p := &api.EndpointAddress{}
-					*p = *epa
-					allAddrs[ak] = p
-				}
-				// Remember that this port maps to this address.
-				if _, found := portsToAddrs[*epp]; !found {
-					portsToAddrs[*epp] = addressSet{}
-				}
-				portsToAddrs[*epp].Insert(allAddrs[ak])
+				joinAddress(&subsets[i].Addresses[k], epp, true, allAddrs, portsToAddrs)
```

Comment:
can the function joinAddress be renamed to something that better reflects what it is doing ?